src/main/java/net/datafaker/providers/base/Barcode.java

```diff
@@ -67,23 +67,24 @@ private long ean(int length) {
             even = odd;
             odd = tmp;
         }
-        int var = 0;
-
-        if (length == 13) {
-            var = odd + even + (even << 1);
-        } else if (length == 8 || length == 14 || length == 12) {
-            var = odd + even + (odd << 1);
-        }
-
-        int rounded = roundToHighestMultiplyOfTen((var));
-        int checkDigit = rounded - var;
+        int output = calculateOutput(length, odd, even);
+        int rounded = roundToHighestMultiplyOfTen((output));
+        int checkDigit = rounded - output;
```

Comment:
```suggestion
        final int output = calculateOutput(length, odd, even);
        final int rounded = roundToHighestMultiplyOfTen((output));
        final int checkDigit = rounded - output;
```
why not final?

Reply:
> why not final?

I did not plan to change the initialization of variables, hence.